src/game/common/skirmishbattlehonors.cpp

```diff
@@ -27,3 +39,617 @@ int SkirmishBattleHonors::Get_Endurance_Medal(Utf8String name, int level) const
+{
+    Utf8String str;
+    str.Format("CHINACampaign_%d", level);
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_USA_Campaign_Complete(int level) const
+{
+    Utf8String str;
+    str.Format("USACampaign_%d", level);
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_GLA_Campaign_Complete(int level) const
+{
+    Utf8String str;
+    str.Format("GLACampaign_%d", level);
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_Challenge_Campaign_Complete(int general, int level) const
+{
+    Utf8String str;
+    str.Format("ChallengeCampaign%d_%d", general, level);
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_Honors() const
+{
+    Utf8String str = "Honors";
+    return Get_Int(str, 0);
+}
+
+void Reset_Battle_Honor_Insertion()
+{
+    g_rowsToSkip = 0;
+}
+
+void Insert_Battle_Honor(GameWindow* listbox, Image const* image, char earned, int tooltip, int* row, int* column, Utf16String str, int score)
+{
+    int height = (int)(((float)g_theDisplay->Get_Height() / 600.0) * 41.0);
+    int width = (int)(((float)g_theDisplay->Get_Width() / 800.0) * 40.0);
+    int color;
+    if (earned) {
+        color = 0xFFFFFFFF;
+    } else {
+        tooltip |= 0x8000000;
+        color = 0xFF505050;
+    }
+    Gadget_List_Box_Add_Entry_Image(listbox, image, *row, *column, height, width, 1, color);
+    Gadget_List_Box_Set_Item_Data(listbox, reinterpret_cast<void *>(tooltip), *row, *column);
+    Gadget_List_Box_Set_Item_Data(listbox, reinterpret_cast<void *>(score), *row - 1, *column);
+    int nextColumn = *column + 1;
+    *column = nextColumn;
+    int numColumns = Gadget_List_Box_Get_Num_Columns(listbox);
+    if (nextColumn >= numColumns) {
+        *column = 0;
+        *row += g_rowsToSkip + 1;
+        numColumns = g_rowsToSkip - 1;
+        if (numColumns >= 0) {
+            g_rowsToSkip = numColumns;
+        } else {
+            g_rowsToSkip = 0;
+        }
+    }
+}
+
+enum SkirmishBattleHonorsTooltips : int32_t
+{
+    STREAK = 1 << 1,
+    LOYALTY_USA = 1 << 5,
+    LOYALTY_CHINA = 1 << 6,
+    BATTLE_TANK = 1 << 7,
+    AIR_WING = 1 << 8,
+    LOYALTY_GLA = 1 << 9,
+    ENDURANCE = 1 << 10,
+    CAMPAIGN_USA = 1 << 11,
+    CAMPAIGN_CHINA = 1 << 12,
+    CAMPAIGN_GLA = 1 << 13,
+    BLITZ_5 = 1 << 14,
+    BLITZ_10 = 1 << 15,
+    FAIR_PLAY = 1 << 16,
+    APOCALYPSE = 1 << 17,
+    OFFICERS_CLUB = 1 << 18,
+    DOMINATION = 1 << 19,
+    CAMPAIGN_CHALLENGE = 1 << 20,
+    ULTIMATE = 1 << 21,
+    GLOBAL_GENERAL = 1 << 22,
+    DOMINATION_ONLINE = 1 << 23,
+    STREAK_ONLINE = 1 << 24,
+    CHALLENGE = 1 << 25,
+};
+
+void Populate_Skirmish_Battle_Honors()
+{
+    NameKeyType key = g_theNameKeyGenerator->Name_To_Key("SkirmishGameOptionsMenu.wnd:ListboxInfo");
+    GameWindow *listbox = g_theWindowManager->Win_Get_Window_From_Id(nullptr, key);
+    if (listbox == NULL) {
+        return;
+    }
+    SkirmishBattleHonors *BattleHonors = nullptr;
```

Comment:
Thyme naming convention for in function variables is `battle_honors`

affects several variable names in this file